libs/api-interfaces/src/lib/aggregate-views/view-models/resources/types/resource-type-to-view-model.ts

```diff
@@ -0,0 +1,26 @@
+import { IBibliographicReferenceViewModel } from '../bibliographic-reference';
+import { IBookViewModel } from '../book';
+import { IMediaItemViewModel } from '../media-items';
+import { IPhotographViewModel } from '../photograph.view-model.interface';
+import { ResourceType } from '../resource-type.enum';
+import { ISongViewModel } from '../song.view-model.interface';
+import { ISpatialFeatureViewModel } from '../spatial-feature';
+import { ITermViewModel } from '../term.view-model.interface';
+import { ITranscribedAudioViewModel } from '../transcribed-audio.view-model.interface';
+import { IVocabularyListViewModel } from '../vocabulary-list';
+
+/**
+ * Can we avoid this? It seems to make our system a little less extensible to
```

Comment:
Consider how we might program to the base `IBaseViewModel` instead. Do we really need type safety wherever we are using this mapped-type for inference? If so, we may just need to keep this.